File: 3.cpp
struct Solution {
    int lengthOfLongestSubstring(string &s) {
        if (s.size() <= 1){
            return s.size();
        }
        int count[128] = {0}, res = 1, i = 0;
        for (int j = 0; j < s.size(); ++j){
            char &c = s[j];
            ++count[c];
            while (count[c] > 1){
                --count[s[i++]];
            }
            res = max(res, j - i + 1);
        }
        return res;
    }
};

File: sol/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) <= 1:
            return len(s)
        count = dict()
        res = 1
        i = 0
        for j, c in enumerate(s):
            count[c] = count.get(c, 0) + 1
            while count[c] > 1:
                count[s[i]] -= 1
                i += 1
            res = max(res, j - i + 1)
        return res
